internal/otc_wrapper.go

```diff
@@ -34,16 +34,21 @@ func NewOtcClientFromConfig(config ConfigStruct) (*OtcWrapper, error) {
 	return &OtcWrapper{providerClient: provider, Region: string(config.AuthenticationData.Region)}, nil
 }
 
-func (c *OtcWrapper) GetMetrics() ([]otcMetrics.MetricInfoList, error) {
+func (c *OtcWrapper) GetMetrics(logger ILogger) ([]otcMetrics.MetricInfoList, error) {
```

Comment:
I think right here we can simplify things. Instead of passing the logger to every method call, we can make it a part of the struct.

Reply:
Oh that's a good idea! Sorry, I'm still not used to using interfaces and such yet!

Reply:
Everything is fine and you don't need to apologise. Try this idea out and see how it looks. :)

Reply:
Yup, that is definitely a much nicer way of doing it!